From 95e694452be6c040bb8040f39b260c97b214f1c0 Mon Sep 17 00:00:00 2001
From: Ralf Engels <ralf-engels@gmx.de>
Date: Wed, 6 Feb 2013 01:02:45 +0100
Subject: [PATCH 057/289] Fix crashed when dynamic playlist based on last.fm
 is aborted

Since the bias solver runs in a different thread care must be taken
for signals.

BUG: 314243
FIXED-IN: 2.8
---
 src/browsers/BrowserMessageArea.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/browsers/BrowserMessageArea.cpp b/src/browsers/BrowserMessageArea.cpp
index 0a2b391..dc36257 100644
--- a/src/browsers/BrowserMessageArea.cpp
+++ b/src/browsers/BrowserMessageArea.cpp
@@ -117,11 +117,11 @@ BrowserMessageArea::newProgressOperation( QObject *sender, const QString &text,
     newBar->setDescription( text );
     newBar->setMaximum( maximum );
     connect( sender, SIGNAL(destroyed( QObject * )), m_progressBar,
-             SLOT(endProgressOperation( QObject * )) );
+             SLOT(endProgressOperation( QObject * )), Qt::QueuedConnection );
     connect( sender, SIGNAL(endProgressOperation( QObject * )), m_progressBar,
-             SLOT(endProgressOperation( QObject * )) );
+             SLOT(endProgressOperation( QObject * )), Qt::QueuedConnection );
     connect( sender, SIGNAL(incrementProgress()), m_progressBar,
-             SLOT(slotIncrementProgress()) );
+             SLOT(slotIncrementProgress()), Qt::QueuedConnection );
     connect( sender, SIGNAL(totalSteps( int )), newBar, SLOT(slotTotalSteps( int )) );
     newBar->setAbortSlot( obj, slot, type );
     m_progressBar->addProgressBar( newBar, sender );
-- 
1.7.11.7 
